contracts/p1/BackingManager.sol

```diff
@@ -146,14 +130,106 @@ contract BackingManagerP1 is TradingP1, IBackingManager {
                     if (req.sellAmount > bal) stRSR.seizeRSR(req.sellAmount - bal);
                 }
 
-                tryTrade(req);
+                openTrade(req);
             } else {
                 // Haircut time
                 compromiseBasketsNeeded(basketsHeld.bottom, basketsNeeded);
             }
         }
     }
 
+    /// Maintain the overall backing policy in an atomic swap via a dutch auction
+    /// @dev Caller must have granted tokenIn allowances for required tokenIn bal
+    /// @param tokenIn The ERC20 token provided by the caller
+    /// @param tokenOut The ERC20 token being purchased by the caller
+    /// @param amountOut {qTokenOut} The exact quantity of tokenOut being purchased
+    /// @return s The exact Swap performed
+    /// @custom:interaction RCEI
+    function swap(
+        IERC20 tokenIn,
+        IERC20 tokenOut,
+        uint256 amountOut
+    ) external returns (Swap memory s) {
+        // == Refresh ==
+        assetRegistry.refresh();
+        furnace.melt();
+
+        requireCanTrade(0);
+
+        // executeSwap if storage auction already exists
+        if (tradeEnd > block.timestamp) {
```

Comment:
since we do this check in multiple places, lets just make a descriptive function for it.  `dutchAuctionExists()` or something like that

Reply:
Good suggestion. Looks like this doesn't overflow contract size, either.